src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexNode.cs

```diff
@@ -419,6 +498,12 @@ internal static SymbolicRegexNode<TSet> CreateBoundaryAnchor(SymbolicRegexBuilde
         internal static SymbolicRegexNode<TSet> CreateLoop(SymbolicRegexBuilder<TSet> builder, SymbolicRegexNode<TSet> body, int lower, int upper, bool isLazy)
         {
             Debug.Assert(lower >= 0 && lower <= upper);
+            //avoid wrapping X? inside (X?)?, or X?? inside (X??)??
```

Comment:
In addition to the comments saying what's being done, it'd be _really_ helpful if they said why it's being done, e.g. what happens if we don't do this?